src/FlatSharp.Compiler.UnityStub/UnityExtensions.cs

```diff
@@ -0,0 +1,89 @@
+﻿/*
+ * Copyright 2022 Unity Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using Unity.Collections;
+#if !FLATSHARP_UNITY_POLYFILLS
+using Unity.Collections.LowLevel.Unsafe;
+#endif
+
+namespace FlatSharp.Internal
+{
+#if !FLATSHARP_UNITY_POLYFILLS
+    unsafe
+#endif
+    public static class UnityExtensions
+    {
+        public static NativeArray<TElement> ReadNativeArrayBlock<TElement, TBuffer>(this TBuffer buffer, int uoffset)
+            where TElement : struct
+            where TBuffer : FlatSharp.IInputBuffer
+        {
+    #if FLATSHARP_UNITY_POLYFILLS
+            throw new NotImplementedException();
+    #else
+            // The local value stores a uoffset_t, so follow that now.
+            uoffset = uoffset + buffer.ReadUOffset(uoffset);
+
+            var mem = buffer.GetReadOnlyByteMemory(uoffset + sizeof(uint), (int)buffer.ReadUInt(uoffset));
+
+            // Create NativeArray around external memory.
+            // The assumption is this is native memory or pinned beyond 'Pin' call below. 
+            using (var pin = mem.Pin())
+            {
+                var na = NativeArrayUnsafeUtility.ConvertExistingDataToNativeArray<TElement>(pin.Pointer, mem.Length, Allocator.Invalid);
```

Comment:
Is `NativeArrayUnsafeUtility` defined somewhere else and the #ifdef's only apply when building not-for-Flatsharp but in some other repo?

Assuming this does a copy, right? One of the things I've tried to avoid in FlatSharp is that when using Lazy deserialization, we try to avoid doing Greedy things like array copies. I've had to part with that restriction recently, but it still makes me grimace a little bit.

I guess with Native memory, you don't have a GC overhead or anything like that, so it's not too bad if the array is small-ish.

Reply:
Yeah, it's defined as part of Unity's APIs.  We should be able to provide a stub.  It converts an existing bare pointer/length to a NativeArray; doesn't do a copy.

Reply:
Stub has been added and is discussed in a different issue.